### PR TITLE
Feature: GCE add fix reuse cluster feature

### PR DESCRIPTION
**Pull request by dkropachev**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by bentsi** [comment]
@fruch why does Dmitry has so many PEP8 issues? I think it some config issue...

**Comment by fruch** [comment]
Seems like you run autopep8 directly, not via pre-commit. So some of the default command line arguments wasn't used. Please use `pre-commit run -a`, it would also give you a change to fix any pylint error

**Comment by fruch** [comment]
@dkropachev can you rebase from master ? so the unit-tests would run ?

**Comment by fruch** [comment]
@bentsi why do we need this backported ?

**Comment by bentsi** [comment]
Usually we backport fixes. let's merge and backport after a while (if somebody will need it).

**Comment by roydahan** [comment]
I don't think we need it backported to 3.1 for sure.
3.2 is nice to have.

**Comment by dkropachev** [comment]
Rebased

**Comment by bentsi** [comment]
Nice!!!

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -42,25 +45,28 @@ def __init__(self, gce_instance, gce_service, credentials, parent_cluster,  # py
...
+            elif "monitor" in self.name and cluster.Setup.KEEP_ALIVE_MONITOR_NODES:
+                self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_MONITOR_NODES)
+                metadata.append('keep-alive')
             self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, ['keep-alive'])
-        if "monitor" in self.name and cluster.Setup.KEEP_ALIVE_MONITOR_NODES:
-            self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_MONITOR_NODES)
-            self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, ['keep-alive'])
+                                     self._instance, metadata, ex_metadata)
```
Comment (amoskong) [inline sdcm/cluster_gce.py]: Hi, @dkropachev  I saw a new err, it seems caused by this change.

```
19:05:28  < t:2019-11-27 11:05:27,866 f:cluster_gce.py  l:104  c:sdcm.cluster         p:ERROR > Node rolling-upgrade-upgrade--centos-db-node-c3a24b48-0-1 [35.231.52.66 | 10.142.0.123] (seed: False): Call to method <bound method GCENodeDriver.ex_set_node_tags of <libcloud.compute.drivers.gce.GCENodeDriver object at 0x7fa218048438>> (retries: 3) failed: ex_set_node_tags() takes 3 positional arguments but 4 were given
```

Comment (amoskong) [reply sdcm/cluster_gce.py]: Reference:
https://libcloud.readthedocs.io/en/latest/compute/drivers/gce.html#libcloud.compute.drivers.gce.GCENodeDriver.ex_set_node_tags
https://libcloud.readthedocs.io/en/latest/_modules/libcloud/compute/drivers/gce.html#GCENodeDriver.ex_set_node_tags

```
ex_set_node_tags(node, tags)

Parameters: 
- node (Node) – Node objecttags (list of str) 
– List of tags to apply to the object
```

Comment (fruch) [reply sdcm/cluster_gce.py]: Seems like we missed that metadata has its own API call

It should be separated into tags, and metadata
https://libcloud.readthedocs.io/en/latest/compute/drivers/gce.html#libcloud.compute.drivers.gce.GCENodeDriver.ex_set_node_metadata

Comment (bentsi) [reply sdcm/cluster_gce.py]: Dmitry is working on fix

**Pull request merged**